src/components/ciPipeline/ciPipeline.service.ts

```diff
@@ -1,16 +1,19 @@
 import { Routes, SourceTypeMap, TriggerType, ViewType } from '../../config'
 import { get, post } from '../../services/api'
-import { CiPipelineSourceTypeBaseOptions } from './ciPipeline.util'
 import { getSourceConfig, getWebhookDataMetaConfig } from '../../services/service'
+import { CiPipelineSourceTypeBaseOptions } from '../CIPipelineN/ciPipeline.utils'
 import { MaterialType, Githost, PatchAction, ScriptType, PluginType, BuildStageType, RefVariableType } from './types'
 
 const emptyStepsData = () => {
     return { id: 0, steps: [] }
 }
 
-export function savePipeline(request): Promise<any> {
-    const URL = `${Routes.CI_PIPELINE}`
-    return post(URL, request)
+export function savePipeline(request, isRegexMaterial = false): Promise<any> {
+    if (isRegexMaterial) {
```

Comment:
Can we write this to return post(`${Routes.CI_PIPELINE_PATCH}${isRegexMaterial?'/regex':''}`, request)